zeth-contracts/contracts/BaseMixer.sol

```diff
@@ -103,15 +127,134 @@ contract BaseMixer is MerkleTreeMiMC7, ERC223ReceivingContract {
     // ============================================================================================ //
     // Reminder: Remember that the primary inputs are ordered as follows:
     // We make sure to have the primary inputs ordered as follow:
-    // [Root, NullifierS, CommitmentS, value_pub_in, value_pub_out]
+    // [Root, NullifierS, CommitmentS, h_sig, h_iS, Residual Field Element(S)]
```

Comment:
Have you considered moving h_sig and everything beyond to before the `NullifierS`?  In this way, the code to reassemble these values shoudl be much simpler (no matter how many inputs / outpus, we can be sure that we can fit the public values and residual parts of elements into a single field element, so we avoid a lot `if (2*size_value+padding*(1+2*jsIn+jsOut) < field_capacity) ....` conditions.

Again, not super important at this stage, but could simplify some of the code.

Reply:
I assumed we had one field element (c.f. next comment).